File: scala-bullets/src/main/java/com/tomergabel/examples/java/ClassWithLogs.java
package com.tomergabel.examples.java;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by tomer on 5/20/14.
 */
public class ClassWithLogs {
    private static Logger log = LoggerFactory.getLogger( ClassWithLogs.class );

    public String getNormalizedName( Person person ) {
        log.info( "getNormalizedName called" );
        log.debug( "Normalizing " + person.toString() );
        String normalizedName = person.getName().toUpperCase().trim();
        log.debug( "Normalized name is: " + normalizedName );
        return normalizedName;
    }
}

File: scala-bullets/src/main/scala/com/tomergabel/examples/bullet4/ClassWithLogs.scala
package com.tomergabel.examples.bullet4

/**
 * Created by tomer on 5/20/14.
 */
class ClassWithLogs extends Logging {

  def getNormalizedName( person: Person ) = {
    logInfo( "getNormalizedName called" )
    logDebug( "Normalizing " + person.toString )
    val normalizedName: String = person.name.toUpperCase.trim
    logDebug( "Normalized name is: " + normalizedName )
  }
}
